plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/stock-indicator/block.tsx

```diff
@@ -56,6 +58,42 @@ const getTextBasedOnStock = ( {
 	return __( 'Out of stock', 'woocommerce' );
 };
 
+/**
+ * Determines whether the stock indicator should be visible.
+ * Stock indicator is not visible when:
+ * - Product type is not allowed (variable, grouped, external)
+ * - Product is in stock, stock is not managed, and not on backorder
+ * - Product is a variation that is on backorder but backorder notification is disabled and stock is managed
+ *
+ * @param {ProductResponseItem} product Product object
+ * @return {boolean} True if stock indicator should be visible
+ */
+const isStockVisible = ( product: ProductResponseItem ): boolean => {
+	const productTypesWithoutStockIndicator = getSetting< string[] >(
+		'productTypesWithoutStockIndicator',
+		[ 'external', 'grouped', 'variable' ]
+	);
+
+	const isProductTypeAllowed = ! productTypesWithoutStockIndicator.includes(
+		product.type
+	);
+
+	const isInStockHidden =
+		product.is_in_stock &&
+		product.manage_stock === false &&
+		! product.is_on_backorder;
+
+	const isStockHiddenForVariation =
+		! product.backorder_notification_enabled &&
```

Comment:
According to the definition in the endpoint, `is_on_backorder` will already be false if notifications are disabled, so I think `backorder_notification_enabled` isn't needed?

Reply:
We will need to know if we notify the customer or not. 

![Screenshot 2024-12-05 at 4 04 42 PM](https://github.com/user-attachments/assets/2104c160-7a60-4dc6-8f89-1036b5aedeee)

`is_on_backorder` will be true for the options: `Allow` and `Allow, but notify customer`, however we won't show the message when the option `Allow` is selected. We need the extra attribute in order to know whether we should show the stock indicator or not. Same check is done [here](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/abstracts/abstract-wc-product.php#L2177).

Reply:
Oh, good catch! `is_on_backorder` returns true even if backorder notifications are disabled. :thinking: 

To me, this looks like a bug in the Store API endpoint. I think `is_on_backorder` should actually check that (1) the product is on backorder and (2) notifications are enabled.

Otherwise, it feels weird that we are exposing these internal settings to the Store API, which is public and available to shoppers. For example, as a merchant, I might sell a product pretending I have it in stock even if I don't have it, so IMO `is_on_backorder` should only be true if the product is on backorder **and** notifications are enabled. What do you think?

I think it's worth considering because it would allow us to avoid introducing the `backorder_notification_enabled` property.